v3io/dataplane/transport/httpclient.py

```diff
@@ -95,37 +93,40 @@ def wait_response(self, request, raise_for_status=None, num_retries=1):
                 # return the response
                 return response
 
-            except self._wait_response_exceptions as e:
+            except v3io.dataplane.response.HttpResponseError as response_error:
+                self._logger.warn_with('Response error: {}'.format(str(response_error)))
+                raise response_error
+            except BaseException as e:
```

Comment:
Ah, the log is actually a bit confusing now because the exception is not necessarily a disconnect, but I added the exception type and message to the log so that the information is there. Should probably improve the message to reflect that.

Regarding why I made this the general "unknown exception" case, it's because a retry can still help in some cases (e.g. SSL error) that are hard to enumerate, and because it wasn't closing and recreating the connection before putting it back into the queue.

Reply:
#92